haskell/src/Client/StablecoinClient/Class.hs

```diff
@@ -0,0 +1,138 @@
+-- SPDX-FileCopyrightText: 2020 tqtezos
+-- SPDX-License-Identifier: MIT
+
+module Client.StablecoinClient.Class
+  ( HasStablecoinClient(..)
+  , DeploymentData(..)
+  ) where
+
+import Lorentz (EntryPointRef(Call), HasEntryPointArg, arg, useHasEntryPointArg)
+import Michelson.Text (MText)
+import Michelson.Typed (Dict(..), IsoValue, toVal)
+import qualified Michelson.Typed as T
+import Morley.Client
+  (AddressOrAlias(..), HasTezosClient(..), HasTezosRpc(..), MorleyClientM, lTransfer,
+  originateUntypedContract, resolveAddress)
+import Morley.Client.Logging (WithClientLog)
+import Tezos.Address (Address)
+import Tezos.Core (unsafeMkMutez)
+import Util.Named ((:!), (.!))
+
+import Client.Contract (mkInitialStorage, parseStablecoinContract)
+import Lorentz.Contracts.Stablecoin (Parameter)
+
+data DeploymentData = DeploymentData
+  { ddMasterMinter :: AddressOrAlias
+  , ddContractOwner :: AddressOrAlias
+  , ddPauser :: AddressOrAlias
+  , ddSafelist :: AddressOrAlias
+  , ddTokenName :: MText
+  , ddTokenSymbol :: MText
+  , ddTokenDecimals :: Natural
+  }
+
+class Monad m => HasStablecoinClient m where
```

Comment:
What is this type class needed for? AFAIU, right now it has only one instance defined right here. Wouldn't it be simpler to have `deploy ::  "sender" :! AddressOrAlias -> DeploymentData -> MorleyClientM Address`? Are you going to add another implementation for tests? That's one possible approach (to mock these methods), but we can have tests for real implementation where we don't mock anything and these tests are more useful. We are successfully doing it in `tezos-btc`. There we started with mock tests because setting up and running tests for real implementation was complicated. But now it's not complicated, we are successfully running nettests on CI. So usefulness of those mock tests seems to be small. And after we do morley's #53 we should be able to write tests using the same interface that will run in emulated environment, on real network using nettest implementation and on real network using `stablecoin-client`.

Reply:
Ah good point! Yeah, I can't think of a scenario where you would run deploy on anything other than `MorleyClientM`, so I'll get rid of the typeclass and simplify this.

Reply:
Done.

Also moved everything under `Stablecoin.*` (e.g. `Stablecoin.Client.Parser`) to mirror the module hierarchy we used in `Morley.Client`.